azure-pipelines.yml

```diff
@@ -0,0 +1,217 @@
+############################################################################################
+#  This Azure Pipelines was created after much time expend exploring the documentation
+#  available online.
+#
+#  Use it at your own risk.
+#
+#  Here is how it works:
+#  - We configure it to run for any branch but only for PRs to master.
+#  - The OFFICIAL_REPO variable avoids issues with Forks.
+#  - The DeployControl job is responsible for defining a variable that is used
+#  by the other deployment jobs to ensure that we only deploy for the OFFICIAL_REPO
+#  and not for PRs.
+#  - We currently use the template to run the tests (See azure-test-template.yml).
+#  - After the tests pass 3 Publish jobs go into action:
+#      - Documentation
+#        This job is executed on commits to master and tags.
+#        It deployes the new docs to the gh-pages branch at GitHub.
+#      - PyPI
+#        This job is executed on tags only.
+#        It deploy the package to the Python Package Index.
+#      - Anaconda
+#        This job is executed on commits to master and tags.
+#        It deploy the package to the DEV or TAG channels at Anaconda Cloud.
+############################################################################################
+
+# Build only for PRs for master branch
+pr:
+  autoCancel: true
+  branches:
+    include:
+      - master
+
+variables:
+  OFFICIAL_REPO: 'slaclab/pydm'
+
+jobs:
+  - template: azure-test-template.yml
+    parameters:
+      name: Linux
+      vmImage: 'Ubuntu 16.04'
+      build_docs: 1
+
+  - template: azure-test-template.yml
+    parameters:
+      name: Windows
+      vmImage: 'vs2017-win2016'
+
+  - template: azure-test-template.yml
+    parameters:
+      name: MacOS
+      vmImage: 'macOS-10.13'
+
+  - job: 'DeployControl'
+    pool:
+      vmImage: 'Ubuntu 16.04'
+    steps:
+    - bash: |
+        echo "##vso[task.setvariable variable=build_ok;isOutput=true]true"
+      name: var
+      condition: |
```

Comment:
The only thing I _really_ don't like about azure pipelines is how these conditions are written, and how they're located after the script block. The best I can come up with to reduce the usage of these is as follows:

Agent variables are translated into environment variables (e.g., in bash it would be `$BUILD_REPOSITORY_NAME` and `$AGENT_OS` instead of `$(Build.Repository.Name)` or `$(Agent.OS)`, respectively). 

Because of that, you might consider squashing some of these conditions where possible and making them into bash conditionals inside the script instead of azure conditionals. That's not something that has to happen in this PR, though, since you worked so hard to get it to this point.

Reply:
I see what you are saying... 
My idea of having this "DeployControl" step was exactly to save this validation to only one variable (build_ok).
I will look into ways to make it better in the future and go with the way you are saying...
This is the first stab at it and I am pretty sure that we will want to revisit it a couple of times before it is 100%...

Reply:
Just to be extra clear, my comment is on how azure does things, not necessarily how you wrote the configuration in this PR. Regardless, the end result completely justifies a bit of syntactic mess (in my opinion). 

Yeah, that seems reasonable regarding the `build_ok` variable. 👍

Reply:
@klauer all is good! I was also complaining about how Azure has this messy environment and conditions... 👍